.eslintrc.js

```diff
@@ -1,110 +1,167 @@
+        // 'react/sort-prop-types'             : ['error', { ignoreCase: true, sortShapeProp: true }],
     },
-  },
-  settings: {
-    react: {
-      version: '16',
+    extends: [
+        'prettier',
+        'prettier/react',
+        'airbnb-base',
+        'binary',
+        'plugin:react/recommended',
+        'plugin:react-hooks/recommended',
+        'plugin:jest-dom/recommended',
+    ],
+    parserOptions: {
+        requireConfigFile: false,
+        ecmaVersion: 8,
+        babelOptions: {
+            presets: ['@babel/preset-react', '@babel/preset-typescript'],
+            plugins: [
+                ['@babel/plugin-proposal-decorators', { legacy: true }],
+                ['@babel/plugin-proposal-class-properties', { loose: true }],
+                '@babel/plugin-proposal-export-default-from',
+                '@babel/plugin-proposal-object-rest-spread',
+                '@babel/plugin-proposal-export-namespace-from',
+                '@babel/plugin-syntax-dynamic-import',
+                '@babel/plugin-proposal-optional-chaining',
+                '@babel/plugin-proposal-nullish-coalescing-operator',
+            ],
+        },
+        ecmaFeatures: {
+            jsx: true,
+        },
     },
-  },
-  overrides: [
-    {
-      files: ['**/__tests__/**/*.[jt]s?(x)', '**/?(*.)+(spec|test).[jt]s?(x)'],
-      extends: ['plugin:testing-library/react'],
+    settings: {
+        react: {
+            version: '16',
+        },
     },
-  ],
+    overrides: [
+        {
+            files: ['**/__tests__/**/*.[jt]s?(x)', '**/?(*.)+(spec|test).[jt]s?(x)'],
+            extends: ['plugin:testing-library/react'],
+        },
+        {
+            files: ['*.ts', '*.tsx'],
+            parser: '@typescript-eslint/parser',
+            plugins: ['@typescript-eslint'],
+            extends: [
+                'plugin:react/recommended',
+                'plugin:@typescript-eslint/recommended',
+                'plugin:prettier/recommended',
+            ],
+            parserOptions: {
+                ecmaversion: 2018,
+                sourceType: 'module',
+                ecmaFeatures: {
+                    jsx: true,
+                },
+                babelOptions: {
+                    presets: ['@babel/preset-react', '@babel/preset-typescript'],
+                    plugins: [
+                        ['@babel/plugin-proposal-decorators', { legacy: true }],
+                        ['@babel/plugin-proposal-class-properties', { loose: true }],
+                        '@babel/plugin-proposal-export-default-from',
+                        '@babel/plugin-proposal-object-rest-spread',
+                        '@babel/plugin-proposal-export-namespace-from',
+                        '@babel/plugin-syntax-dynamic-import',
+                        '@babel/plugin-proposal-optional-chaining',
+                        '@babel/plugin-proposal-nullish-coalescing-operator',
+                    ],
+                },
```

Comment:
If we were to declare babel options for the root package, can we remove the individual package configs as a cleanup measure